File: public/dsa/linked_lists/delete_node_from_linked_list/approach/delete_node_from_linked_list.cpp
#include <bits/stdc++.h>
using namespace std;

// defs-start
struct ListNode {
    int data;
    ListNode* next;
    ListNode(int data, ListNode* next = nullptr) : data(data), next(next) {}
};
// defs-end

void deleteNode(ListNode* node) {
    node->data = node->next->data;
    node->next = node->next->next;
}

// tests-start
int main() {
    ListNode* node = new ListNode(2, new ListNode(3, new ListNode(4)));
    ListNode* head = new ListNode(1, node);

    deleteNode(node);

    ListNode* temp = head;
    while (temp) {
        cout << temp->data << " ";
        temp = temp->next;
    }
    cout << endl;
    return 0;
}
// tests-end

/*output
1 3 4
*/

File: public/dsa/linked_lists/delete_node_from_linked_list/approach/delete_node_from_linked_list.py
# defs-start
class ListNode:
    def __init__(self, data=0, next=None):
        self.data = data
        self.next = next
# defs-end


class Solution:
    def deleteNode(self, node: ListNode) -> None:
        node.data = node.next.data
        node.next = node.next.next


# tests-start
node = ListNode(2, ListNode(3, ListNode(4)))
head = ListNode(1, node)
sol = Solution()
sol.deleteNode(node)
temp = head
while temp:
    print(temp.data, end=' ')
    temp = temp.next
# tests-end

'''output
1 3 4
'''
